crates/api/src/exe/err.rs

```diff
@@ -0,0 +1,61 @@
+#[derive(Debug)]
+pub enum Error {
+    Io(std::io::Error),
+    Reqwest(reqwest::Error),
+    IamGroot(iamgroot::jsonrpc::Error),
+    StarknetApi(starknet_api::StarknetApiError),
+    State(blockifier::state::errors::StateError),
+    EntryPoint(blockifier::execution::errors::EntryPointExecutionError),
+    Transaction(blockifier::transaction::errors::TransactionExecutionError),
+    Custom(&'static str),
+}
```

Comment:
I'm slightly concerned that always mapping by type may not help during debugging: it's good to know an error is an `std::io::Error` but it's crucial to know where it happens.
WIth such a mapping, I fear we may lack this info in the case of multiple places potentially throwing `std::io::Error`.

Reply:
Such mapping is way out of scope of this enum that only defines the error.
What you've described is a `Context`, which can be added as a layer on top with eyre/anyhow.

Reply:
It's not an `anyhow`/`eyre`-like context, the mapping to an `anyhow` error would happen subsequently, once this pieces of info are already lost.
It's fine for now, but I've used this pattern multiple times and systematically complained about it as soon as there are multiple sources of an IO error or transport error or anything.